blueocean-dashboard/src/main/js/components/karaoke/components/LogConsole.jsx

```diff
@@ -13,33 +13,38 @@ const RERENDER_DELAY = 17;
 
 const logger = logging.logger('io.jenkins.blueocean.dashboard.karaoke.LogConsole');
 
-const LogLine = ({ prefix, line, index, router, location }) => {
-    const tokenized = tokenizeANSIString(line);
-    const lineChunks = makeReactChildren(tokenized);
+class LogLine extends PureComponent {
```

Comment:
Why `PureComponent`?

Reply:
The default implementation of shouldComponentUpdate is useful in this scenario:

> If your React component’s render() function renders the same result given the same props and state, you can use React.PureComponent for a performance boost in some cases.
> https://reactjs.org/docs/react-api.html#reactpurecomponent